### DISCUSSION TITLE
0.10.0

### DISCUSSION THREAD

**Discussion by Blyron** [carla-simulator/carla#8496] [Annoucement]
[0.10.0](https://github.com/carla-simulator/carla/releases/tag/0.10.0)

Release 0.10.0
[Ubuntu] [CARLA_0.10.0.tar.gz](https://tiny.carla.org/carla-0-10-0-linux)
[Windows] [CARLA_0.10.0.zip](https://tiny.carla.org/carla-0-10-0-windows)

Changes
* Unreal Engine migration from version 4.26 to version 5.5
    - Nanite enabled for entire Town 10
    - Lumen powers new realistic illumination
    - Improved lighting with new post-process configuration
    - Physics migrated from PhysX to Chaos
* Remodeled Town 10:
    - New buildings
    - Road topology includes uneven surface, potholes, speed bumps
* Remodeled 13 vehicles with more detail:
    - Nissan Patrol
    - Dodge Charger
    - Dodge Charger Police
    - Ford Crown Taxi
    - Lincoln MKZ
    - Mini Cooper
    - Firetruck
    - CarlaCola
    - Ambulance
    - Mercedes Sprinter
    - Mitsubishi Fusorosa bus
    - Mining earth mover (provided by Synkrotron)
* Rework of build system to use cmake
* Native server support for ROS2 (without CARLA-ROS bridge)
* Refactor of semantic segmentation and instance segmentation sensors
    - Semantic segmentation can now be visualized in UE editor for debugging
    - Only supports meshes with Nanite enabled
* Included support to data-driven realistic traffic generation powered by InvertedAI
* Updated Python API scripts with extra examples in the PythonAPI folder:
    - examples/invertedai_traffic.py
    - examples/interpolate_camera.py
    - examples/vehicle_doors_demo.py
    - examples/vehicle_lights_demo.py
    - examples/visualize_radar.py
    - examples/ros2_native.py
    - util/tools/generate_video_from_frames.py
    - util/apply_texture.py
    - util/change_map_layer.py
    - util/manage_environment_objects.py
    - util/manage_traffic_light.py
    - util/recorder_comparer.py
    - util/show_crosswalks.py
    - util/show_junctions.py
    - util/show_topology.py
    - util/spectator_transform.py
* Motion blur disabled below 1080p resolution to avoid artefacts
* Python API now supports Python 3.8, 3.9, 3.10, 3.11 and 3.12
* Python API support dropped for Python versions 3.7 and lower
* Added support for Scenic 3.0
* RSS functionality removed from docs
* Removed Light Manager from API and docs
* Added Mine01 off-road mining map from Synkrotron

<hr /><em>This discussion was created from the release <a href='https://github.com/carla-simulator/carla/releases/tag/'>0.10.0</a>.</em>

**Comment by Honghe** [comment]
@Blyron 
The  example of `util/change_map_layer.py` doesn't work, and the [release note](https://carla.org/2024/12/19/release-0.10.0/) says `Map layers API is not supported`, so what's the use of this updated example?

**Reply by Blyron** [reply]
We need to update it and update the functionality

**Comment by ydavs** [comment]
"Semantic segmentation can now be visualized in UE editor for debugging". How can I visualize the output of semantic segmentation in editor?

**Reply by Blyron** [reply]
vis SegmentationBuffer in console command
or 
 vis SegmentationBuffer uv0  to complete screen seen in segmentation buffer mode.
 
 Remember Segmentation buffer only works in play mode.
